gio-macros/src/lib.rs

```diff
@@ -0,0 +1,63 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+extern crate proc_macro;
+
+mod action_map;
+
+use proc_macro::TokenStream;
+use syn::{parse_macro_input, ItemImpl};
+
+/// Macro for creating [`gio::Action`]s and registering them in a given [`gio::ActionMap`].
+/// It generates a method `register_actions` with a following signature:
+///
+/// ```rust
+/// # struct X; impl X {
+/// fn register_actions<AM: glib::object::IsA<gio::ActionMap>>(&self, map: &AM)
+/// # {} }
+/// ```
+///
+/// # Example
+/// ```rust
+/// use gio::prelude::*;
+///
+/// #[derive(glib::Downgrade)]
+/// pub struct MyApplication(gio::Application);
+///
+/// impl MyApplication {
+///     pub fn new() -> Self {
+///         let app = Self(gio::Application::new(
+///             Some("com.example.MyApplication"),
+///             gio::ApplicationFlags::FLAGS_NONE,
+///         ));
+///         app.register_actions(&app.0);
+///         app
+///     }
+/// }
+///
+/// #[gio::actions]
+/// impl MyApplication {
+///     fn action1(&self) {
```

Comment:
```suggestion
///     #[action]
///     fn action1(&self) {
```

Would probably be better than taking any and every function and turning it into an action

Reply:
They *are* actions. All methods inside this `impl` item are actions. It is an `impl` item who is annotated and produces an `ActionMap`. E.g. it is possible to have more than one `ActionMap` implemented by a single type.

Reply:
I guess that's why I'm asking for a proper example that makes use of most of the features of this :)

Reply:
There are two :) One in a doc-test and another in tests.

I also mentioned one in #240. [There is a regular `impl` block with private methods and one more `impl` block with actions](https://github.com/andy128k/find_duplicates/blob/master/find_duplicates_gui/src/main_window.rs#L377).

Reply:
I meant a meaningful example how this would be used in practice as part of the examples or examples4 subdirectory.

Or maybe the docs and docs example just has to be extended to cover these questions too.

Reply:
Sure. I will create an example with some realistic actions.